### ISSUE TITLE
Release 4.0.0

### ISSUE THREAD

**Issue by tzugen** [ogarcia/ultrasonic#730]
I have contacted the Google Play Store team via Email.
This has finally moved the current update out of review.
Unfortunately it was still rejected, because the 3.0.0 update still contained the Paypal link.

I am unsure if 3.1.1 contains the rebuild about screen. If yes we should submit it now to the Play Store.
Otherwise we should make a new release from trunk and release it to Play Store.

**Comment by ogarcia** [comment]
I don't know why, but they have tested version 3.0.0 instead of 3.1.1. The change in the about was made on #684 that was included in version 3.1.0.

I have sent an appeal through the google play console notifying them of the error. I do not have much hope that they will listen to us, let's see what happens.

In any case, I think that once #726 is ready we should release a new version which should force a new revision that, I hope, will correct the problem.

**Comment by ogarcia** [comment]
Speaking of #726 it's a pretty big change. Once we release it, should we change the major version and release version 4.0.0?

**Comment by tzugen** [comment]
I think we could try to just relabel 3.1.1 as 3.1.2 and hand it in for review as well. The appeal might take ages, could be faster to just try to get a "new" version reviewed..

**Comment by ogarcia** [comment]
> I think we could try to just relabel 3.1.1 as 3.1.2 and hand it in for review as well. The appeal might take ages, could be faster to just try to get a "new" version reviewed..

Lately they are taking as long to review appeals as new versions. Keep in mind that 3.1.1 was uploaded on February 17 and it took them until April 7 to review it! Almost two months!

On the other hand I don't think it is worth re-labeling 3.1.1 as 3.1.2 because we would have to include some change and it gives us more work than the benefit.

IMHO, I think the best thing to do is to focus on 4.0.0 and when we have it we will upload it and see if this time we have more luck.

**Comment by tzugen** [comment]
OK !

**Comment by tzugen** [comment]
But should we release a minor version before to F Droid? Because the database will change versions, and otherwise there is no downgrade path from version 4.0

**Comment by ogarcia** [comment]
> But should we release a minor version before to F Droid? Because the database will change versions, and otherwise there is no downgrade path from version 4.0

I think you want to merge #720 in 4.0. Yes, we can release a 3.2.0 to avoid database problems. What do you think if we simply extract the changes in AppDatabase.kt to a new branch to merge it with the master and thus have that bridge version? Or if you prefer to add something else it is also possible. Please prepare a new pull request with what you want to bring to this new intermediate version and we will release it without any problem.

**Comment by tzugen** [comment]
No, our current HEAD is the bridge version :) It has the migration code from 5->4

**Comment by ogarcia** [comment]
You say to make version 3.2.0 with what we currently have in the develop branch?

**Comment by tzugen** [comment]
Yes, that's what I meant :)

**Comment by ogarcia** [comment]
[Release 3.2.0](https://github.com/ultrasonic/ultrasonic/releases/tag/3.2.0) done :wink: 

Let's maintain this issue open for 4.0.0

**Comment by ogarcia** [comment]
Shit, CircleCI is failing building the signed apk. I'll pick your changes in `.circleci/config.yml` from #726 and try again.

**Comment by ogarcia** [comment]
Finally! CircleCI has been a bit of a pain in the ass, but the version has been released. I have also tried it on Google Play, let see what happens.

**Comment by tzugen** [comment]
Google Play 3.1.1 and 3.2.0 are approved :)

**Comment by tzugen** [comment]
We are moving to Gitlab. Please continue the discussion there: https://gitlab.com/ultrasonic/ultrasonic/-/issues/730
